### PR TITLE
Sync translated resource strings to translations data model

### PR DESCRIPTION
**Pull request by camallen**
Batch process to seed the conversion of translatable resources to new translations data model. This will be run in a rake task to initially seed the data / iron out schema bugs. 

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/models/concerns/translation_strings.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  end
+
+  def removed_nested_attributes(attributes_to_slice, key)
+    attributes[key.to_sym].map do |nested_object|
+      nested_object.slice(*attributes_to_slice)
+    end
+  end
+
+  def flatten_nested_schema(hash_schema)
+    # TODO: convert this to a visitor class that task the schema and converts
```
Comment (camallen) [inline app/models/concerns/translation_strings.rb]: I'd like to get this in but i'm also happy to do this work before merging.

File: app/models/translation.rb [hunk 2]
```diff
@@ -4,9 +4,12 @@ class Translation < ActiveRecord::Base
   validates_presence_of :language
 
   # TODO: add a unique validation for translated_type, id, language
+  # and a unique index
```
Comment (camallen) [inline app/models/translation.rb]: I'll do these model updates in another PR

File: app/models/concerns/translation_strings.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  # pandora app uses the tasks key to find the tasks strings
+  def transform_workflow_attributes
+    attributes["tasks"] = attributes.delete("strings")
+  end
+
+  def transform_tutorial_attributes(nested_key="steps")
+    attributes[nested_key] = removed_nested_attributes(%i(content), nested_key)
+  end
+
+  def removed_nested_attributes(attributes_to_slice, key)
```
Comment (camallen) [inline app/models/concerns/translation_strings.rb]: I don't think this method name is appropriate, perhaps it should be `extract_nested_attributes`

Comment (marten) [reply app/models/concerns/translation_strings.rb]: Agree

**Pull request merged**